packages/bitcore-node/src/services/pruning.ts

```diff
@@ -112,144 +121,173 @@ export class PruningService {
+                try {
+                  const nodeTx: RPCTransaction = await this.rpcs[`${chain}:${network}`].getTransaction(tx.txid);
+                  if (nodeTx) {
+                    logger.warn(`Tx ${tx.txid} is still in the mempool${VERBOSE ? ': %o' : ''}`, nodeTx);
+                    return cb();
+                  }
+                } catch (err: any) {
+                  if (err.code !== -5) { // -5: No such mempool or blockchain transaction. Use gettransaction for wallet transactions.
+                    logger.error(`Error checking tx ${tx.txid} in the mempool: ${err.message}`);
+                    return cb();
+                  }
                 }
-                if (coin.spentTxid) {
-                  spentTxids.add(coin.spentTxid);
-                  if (spentTxids.size > DESCENDANT_LIMIT) {
-                    logger.warn(`${tx.txid} has too many decendants`);
+                logger.info(`Finding ${tx.txid} outputs and dependent outputs`);
+                const outputGenerator = this.transactionModel.yieldRelatedOutputs(tx.txid);
+                let spentTxids = new Set<string>();
+                for await (const coin of outputGenerator) {
+                  if (coin.mintHeight >= 0 || coin.spentHeight >= 0) {
+                    logger.error(`Cannot prune coin! ${coin.mintTxid}`);
                     return cb();
                   }
+                  if (coin.spentTxid) {
+                    spentTxids.add(coin.spentTxid);
+                    if (spentTxids.size > DESCENDANT_LIMIT) {
+                      logger.warn(`${tx.txid} has too many decendants`);
+                      return cb();
+                    }
+                  }
                 }
+                spentTxids.add(tx.txid);
+                rmCount += spentTxids.size;
+                const uniqueTxids = Array.from(spentTxids);
+                await this.removeOldMempool(chain, network, uniqueTxids);
+                logger.info(`Removed tx ${tx.txid} and ${spentTxids.size - 1} dependent txs`);
+                return cb();
               }
-              spentTxids.add(tx.txid);
-              rmCount += spentTxids.size;
-              const uniqueTxids = Array.from(spentTxids);
-              await this.removeOldMempool(chain, network, uniqueTxids);
-              logger.info(`Removed tx ${tx.txid} and ${spentTxids.size - 1} dependent txs`);
-              return cb();
-            }
-          })
-        )
-        .on('finish', resolve)
-        .on('error', reject);
-    });
-    logger.info(`Removed all pending txs older than ${days} days: ${rmCount}`);
+            })
+          )
+          .on('finish', resolve)
+          .on('error', reject);
+      });
+      logger.info(`Removed all pending ${chain}:${network} txs older than ${days} days: ${rmCount}`);
+      this.lastRunTimeOld = Date.now();
+    } catch (err: any) {
+      logger.error(`Error processing old mempool txs: ${err.stack || err.message || err}`);
```

Comment:
This method is always called from w/in a try block. Is the behavior that you want to not re-throw? Seems like you'll miss information this way.